drivers/gpio/gpio_max14906.h

```diff
@@ -0,0 +1,316 @@
+/*
+ * Copyright (c) 2024 Analog Devices Inc.
+ * Copyright (c) 2024 Baylibre SAS
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define MAX14906_FAULT2_ENABLES 5
+#define MAX14906_CHANNELS 4
+#define MAX14916_CHANNELS 8
+#define MAX149x6_MAX_PKT_SIZE 3
+
+#define MAX14906_SETOUT_REG		0x0
+#define MAX14906_SETLED_REG		0x1
+#define MAX14906_DOILEVEL_REG		0x2
+#define MAX14906_INT_REG		0x3
+#define MAX14906_OVR_LD_REG		0x4
+#define MAX14906_OPN_WIR_FLT_REG	0x5
+#define MAX14906_SHT_VDD_FLT_REG	0x6
+#define MAX14906_GLOB_ERR_REG		0x7
+#define MAX14906_OPN_WR_EN_REG		0x8
+#define MAX14906_SHT_VDD_EN_REG		0x9
+#define MAX14906_CONFIG1_REG		0xA
+#define MAX14906_CONFIG2_REG		0xB
+#define MAX14906_CONFIG_DI_REG		0xC
+#define MAX14906_CONFIG_DO_REG		0xD
+#define MAX14906_CONFIG_CURR_LIM	0xE
+#define MAX14906_CONFIG_MASK		0xF
+
+#define MAX149x6_CHIP_ADDR_MASK		GENMASK(7, 6)
+#define MAX149x6_ADDR_MASK		GENMASK(4, 1)
+#define MAX149x6_RW_MASK		BIT(0)
+
+/* DoiLevel register */
+#define MAX14906_DOI_LEVEL_MASK(x)	BIT(x)
+
+/* SetOUT register */
+#define MAX14906_HIGHO_MASK(x)		BIT(x)
+
+#define MAX14906_DO_MASK(x)		(GENMASK(1, 0) << (2 * (x)))
+#define MAX14906_CH_DIR_MASK(x)		BIT((x) + 4)
+#define MAX14906_CH(x)			(x)
+#define MAX14906_IEC_TYPE_MASK		BIT(7)
+#define MAX14906_CL_MASK(x)		(GENMASK(1, 0) << (2 * (x)))
+
+/**
+ * @brief Hardwired device address
+ */
+enum max149x6_spi_addr {
+	MAX14906_ADDR_0, /* A0=0, A1=0 */
+	MAX14906_ADDR_1, /* A0=1, A1=0 */
+	MAX14906_ADDR_2, /* A0=0, A1=1 */
+	MAX14906_ADDR_3, /* A0=1, A1=1 */
+};
+
+enum max14906_iec_type {
+	MAX14906_TYPE_1_3,
+	MAX14906_TYPE_2,
+};
+
+/**
+ * @brief Channel configuration options.
+ */
+enum max14906_function {
+	MAX14906_OUT,
+	MAX14906_IN,
+	MAX14906_HIGH_Z
+};
+
+/**
+ * @brief Configuration options for the output driver (on each channel).
+ */
+enum max14906_do_mode {
+	MAX14906_HIGH_SIDE,
+	MAX14906_HIGH_SIDE_INRUSH,
+	MAX14906_PUSH_PULL_CLAMP,
+	MAX14906_PUSH_PULL
+};
+
+/**
+ * @brief Current limit options for output channels.
+ */
+enum max14906_cl {
+	MAX14906_CL_600,
+	MAX14906_CL_130,
+	MAX14906_CL_300,
+	MAX14906_CL_1200,
+};
+
+union max14906_doi_level {
+	uint8_t reg_raw;
+	struct {
+		uint8_t VDDOK_FAULT1:1; /* BIT0 */
+		uint8_t VDDOK_FAULT2:1;
+		uint8_t VDDOK_FAULT3:1;
+		uint8_t VDDOK_FAULT4:1;
+		uint8_t SAFE_DAMAGE_F1:1;
+		uint8_t SAFE_DAMAGE_F2:1;
+		uint8_t SAFE_DAMAGE_F3:1;
+		uint8_t SAFE_DAMAGE_F4:1; /* BIT7 */
+	} reg_bits;
```

Comment:
It is obvious that bitfield and union simplify driver implementation.
I wonder does the zephyr encourage using bitfield @MaureenHelm 

https://github.com/zephyrproject-rtos/zephyr/issues/4009
https://github.com/zephyrproject-rtos/zephyr/pull/4254